### ISSUE TITLE
Torch, tensorflow dependencies

### ISSUE THREAD

**Issue by hop-soellingeraj** [diffgram/python-sdk#11]
Is it possible to remove torch and tensorflow as dependencies to this module?  All I want to do is query the latest labels in a project / dataset.  Do I really need torch and tensorflow?  Thanks!

**Comment by anthony-chaudhary** [comment]
Hey @hop-soellingeraj   Yep! 

You can install with no-dependencies flag -> `pip install diffgram --no-dependencies` 

And to better inform user we can do something like this for pytorch
https://github.com/diffgram/python-sdk/pull/12/files

Does that fit your goal? Tell me if I'm misunderstanding your intent

**Comment by anthony-chaudhary** [comment]
@PJEstrada In this context - what do you think of this? https://github.com/diffgram/python-sdk/pull/12/commits/7015f69dc31a1f3fd290a135289a23042d29d11c

**Comment by PJEstrada** [comment]
Hey guys, Yes I think that's a good idea @anthony-sarkis, It's good to think that nor everyone will be using those libraries given that they are used for a specific feature of the SDK

**Comment by anthony-chaudhary** [comment]
Thanks Pablo!!

@hop-soellingeraj   I'll await your feedback but otherwise can merge this then

**Comment by hop-soellingeraj** [comment]
> Hey @hop-soellingeraj Yep!
> 
> You can install with no-dependencies flag -> `pip install diffgram --no-dependencies`
> 
> And to better inform user we can do something like this for pytorch
> https://github.com/diffgram/python-sdk/pull/12/files
> 
> Does that fit your goal? Tell me if I'm misunderstanding your intent

I saw a similar message with tensorflow.  The UX I think is best is to install no dependencies for the base functionality, but when the advanced features which require tf or torch are requested, then it gives a prescriptive message like `Additional functionality is needed to run this function run pip install diffgram[tensorflow]` or whatever the package is.  The package DVC  has an example of this in their client lib

Here's the doc for that suggestion:
> If you installed DVC via pip and plan to use cloud services as remote storage, you might need to install these optional dependencies: [s3], [azure], [gdrive], [gs], [oss], [ssh]. Alternatively, use [all] to include them all. The command should look like this: pip install "dvc[s3]". (This example installs boto3 library along with DVC to support S3 storage.)
https://dvc.org/doc/command-reference/remote/add

**Comment by anthony-chaudhary** [comment]
That sounds great!! I'm going to close this for now becuase we will merge the related branch which does make them optional. We can do a seperate issue for improving the user experience regarding which ones to install.

Thank you very much!!
